src/contexts/registration/RegistrationContext.js

```diff
@@ -0,0 +1,52 @@
+import React, { createContext, useState } from 'react';
+
+const RegistrationContext = createContext();
+
+export const RegistrationState = props => {
+  const [user, setUser] = useState({
+    email: '',
+    password: '',
+    passwordRepeat: ''
+  });
+
+  const [tooltip, setTooltip] = useState({
+    emailTooltip: false,
+    passwordTooltip: false,
+    passwordRepeatTooltip: false
+  });
+
+  const [termsAccepted, setTermsAccepted] = useState(false);
+
+  const { email, password, passwordRepeat } = user;
+  const { emailTooltip, passwordTooltip, passwordRepeatTooltip } = tooltip;
+
+  const onTextChanged = e => {
+    setTooltips(e.target);
+    setUser({ ...user, [e.target.id]: e.target.value });
+  };
+
+  const setTooltips = target => {
+    if (target.validity.patternMismatch || target.validity.typeMismatch) {
+      setTooltip({ ...tooltip, [target.id + 'Tooltip']: true });
+    } else setTooltip({ ...tooltip, [target.id + 'Tooltip']: false });
+  };
```

Comment:
Skoro i tak się męczymy z tooltipami (możliwie będziemy je mocno zmieniać), to może cała ich logika i generalnie same tooltipy warto wrzucić do osobnego komponentu?